app/src/main/java/com/sweng/scopehud/SettingsActivity.java

```diff
@@ -0,0 +1,237 @@
+package com.sweng.scopehud;
+
+import android.app.AlertDialog;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.auth.UserProfileChangeRequest;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class SettingsActivity extends AppCompatActivity {
```

Comment:
This should extend the NavigationActivity rather than the AppCompatActivity, and execute the necessary setup in onCreate() (just like MainActivity for example)